Hazel/src/Platform/Windows/WindowsPlatformUtils.cpp

```diff
@@ -40,12 +41,25 @@ namespace Hazel {
 		ofn.nMaxFile = sizeof(szFile);
 		ofn.lpstrFilter = filter;
 		ofn.nFilterIndex = 1;
+
+		// Sets the default extension by extracting it from the filter
+		std::string filterString(filter);
+		std::stringstream s(filterString);
+		std::string defaultExtension;
+
+		while (s >> defaultExtension) {
+			if (defaultExtension.find('*') < defaultExtension.length()) {
+				ofn.lpstrDefExt = defaultExtension.c_str();
+			}
+		}
+		//
+
```

Comment:
Because each individual filter is null terminated, and assuming the first filter is the default filter, you could replace everything with 
```
ofn.lpstrDefExt = strchr(filter, '\0') + 1;
```